pkg/pillar/cmd/downloader/mdns.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+// github.com/grandcat/zeroconf: under MIT License
+
+package downloader
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/grandcat/zeroconf"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+// given a local server URL string, and returns the modified url with server IP address in it,
+// and bridge name and source IP address for connecting to the server.
+func findDSmDNS(dctx *downloaderContext, serverURL string) (string, string, net.IP, error) {
+	dsURL, err := url.Parse(serverURL)
+	if err != nil {
+		log.Errorf("findDSmDNS: error %v", err)
+		return "", "", nil, err
+	}
+	hostname := dsURL.Hostname()
+	var newURL string
+	var foundIP []net.IP
+
+	urls := strings.Split(serverURL, hostname)
```

Comment:
Calling this "urls" is quite confusing. It is the parts of the URL (excluding the hostname).
So "parts" or something would make more sense.

What about a case where the hostname is "bob" and the serverURL is http://bob/home/bob/file?
That seems like a valid URL to me.
Would SplitN with 1 make more sense to just remove the first instance?